src/test/scala/io/qbeast/sources/QbeastOptionsTest.scala

```diff
@@ -16,94 +16,130 @@
+    options.columnsToIndex shouldBe Seq("id", "name")
   }
 
-  it should "be able to create a QbeastOptions instance with cube size" in withSpark { _ =>
-    val options = QbeastOptions(Map("columnsToIndex" -> "id", "cubeSize" -> "10"))
+  it should "initialize with cubeSize" in withSpark { _ =>
+    val options = QbeastOptions(Map(COLUMNS_TO_INDEX -> "id", CUBE_SIZE -> "10"))
     options.cubeSize shouldBe 10
   }
 
-  it should "initialize with default cube size" in withSpark { _ =>
-    val options = QbeastOptions(Map("columnsToIndex" -> "id"))
-    options.cubeSize shouldBe config.DEFAULT_CUBE_SIZE
+  it should "initialize with tableFormat" in withSpark { _ =>
+    val options = QbeastOptions(Map(COLUMNS_TO_INDEX -> "id", TABLE_FORMAT -> "someFormat"))
+    options.tableFormat shouldBe "someFormat"
   }
 
-  it should "initialize txnAppId and txnVersion" in withSpark { _ =>
-    val options =
-      QbeastOptions(Map("columnsToIndex" -> "id", "txnAppId" -> "app", "txnVersion" -> "version"))
-    options.txnAppId shouldBe Some("app")
-    options.txnVersion shouldBe Some("version")
-  }
-
-  it should "initialize userMetadata" in withSpark { _ =>
-    val options = QbeastOptions(Map("columnsToIndex" -> "id", "userMetadata" -> "metadata"))
-    options.userMetadata shouldBe Some("metadata")
+  it should "initialize with columnStats" in withSpark { _ =>
+    val options = QbeastOptions(
+      Map(COLUMNS_TO_INDEX -> "id", COLUMN_STATS -> """{"col1_min":1",col1_max":10}"""))
+    options.columnStats shouldBe Some("""{"col1_min":1",col1_max":10}""")
   }
 
-  it should "initialize mergeSchema" in withSpark { _ =>
-    val options = QbeastOptions(Map("columnsToIndex" -> "id", "mergeSchema" -> "true"))
-    options.mergeSchema shouldBe Some("true")
+  it should "initialize additional options" in withSpark { _ =>
+    val options =
+      QbeastOptions(
+        Map(
+          COLUMNS_TO_INDEX -> "id",
+          TXN_APP_ID -> "app",
+          TXN_VERSION -> "1",
+          USER_METADATA_OPTION -> "Some metadata",
+          MERGE_SCHEMA_OPTION -> "true",
+          OVERWRITE_SCHEMA_OPTION -> "true"))
+    options.extraOptions(TXN_APP_ID) shouldBe "app"
+    options.extraOptions(TXN_VERSION) shouldBe "1"
+    options.extraOptions(USER_METADATA_OPTION) shouldBe "Some metadata"
+    options.extraOptions(MERGE_SCHEMA_OPTION) shouldBe "true"
+    options.extraOptions(OVERWRITE_SCHEMA_OPTION) shouldBe "true"
+
+    options.columnsToIndex shouldBe Seq("id")
+    options.cubeSize shouldBe DEFAULT_CUBE_SIZE
+    options.tableFormat shouldBe DEFAULT_TABLE_FORMAT
+    options.columnStats shouldBe None
   }
 
-  it should "initialize overwriteSchema" in withSpark { _ =>
-    val options = QbeastOptions(Map("columnsToIndex" -> "id", "overwriteSchema" -> "true"))
-    options.overwriteSchema shouldBe Some("true")
+  it should "create a proper DeltaOptions with extraOptions" in withSpark { spark =>
```

Comment:
Delta-Related tests should be on the delta package

Reply:
Done